### PR TITLE
Update

### PR DESCRIPTION
**Pull request by Jinjiang**
Ref: https://github.com/vuejs/vuejs.org/compare/242dea4e...1a736840

except for:
- https://github.com/vuejs/vuejs.org/commit/42c313c8fd0ae84db48fc1287b9d22903d04df67 (skipped)
- https://github.com/vuejs/vuejs.org/commit/0c1ff02161403382e5139e9f4ae56d1e4474e2ba All videos in Vue Mastery are deployed on Vimeo which is blocked in CN. And also the script from vimeo.com would probably block the page loading. So I didn't sync this change. @yyx990803 is that OK to drop this commit here?
- https://github.com/vuejs/vuejs.org/commit/67766e4f44f5bdb752e2fd6b217ed213f39a3bac I made some changes according to 
https://github.com/vuejs/vuejs.org/pull/1850#discussion_r252132980

Thanks.

### CODE REVIEW

**Comment by Jinjiang** [comment]
Also, some of the images in `/theme/vue/source/images` has been removed because they seem not useful anymore:

- `coding-net-ad.jpg`
- `fancygrid.png`
- `gitee_ad.png`
- `geekbang.png`

@yyx990803 you may need to have a check because it's sponsors related.

Thanks.

File: src/v2/guide/single-file-components.md [hunk 1]
```diff
@@ -62,12 +62,10 @@ order: 402
 
 - **Modern JavaScript with ES2015/16**：阅读 Babel 的 [Learn ES2015 guide](https://babeljs.io/docs/learn-es2015/)。你不需要立刻记住每一个方法，但是你可以保留这个页面以便后期参考。
 
-在你花一天时间了解这些资源之后，我们建议你参考 [webpack](https://github.com/vuejs-templates/webpack) 模板。只要遵循指示，你就能很快地运行一个用到 `.vue` 组件，ES2015 和热重载 (hot-reloading) 的 Vue 项目!
-
-想学习更多 webpack 的知识，请移步[它们的官方文档](https://webpack.js.org/configuration/)以及 [webpack learning academy](https://webpack.academy/p/the-core-concepts)。在 webpack 中，每个模块被打包到 bundle 之前都由一个相应的“loader”来转换，Vue 也提供 [vue-loader](https://github.com/vuejs/vue-loader) 插件来执行 `.vue` 单文件组件 的转换。
+在你花一天时间了解这些资源之后，我们建议你参考 [Vue CLI 3](https://cli.vuejs.org/)。只要遵循指示，你就能很快地运行一个带有 `.vue` 组件、ES2015、webpack 和热重载 (hot-reloading) 的 Vue 项目!
```
Comment (Justineo) [inline src/v2/guide/single-file-components.md]:
```suggestion
在你花一天时间了解这些资源之后，我们建议你参考 [Vue CLI 3](https://cli.vuejs.org/zh/)。只要遵循指示，你就能很快地运行一个带有 `.vue` 组件、ES2015、webpack 和热重载 (hot-reloading) 的 Vue 项目!
```

File: src/v2/guide/single-file-components.md [hunk 2]
```diff
@@ -62,12 +62,10 @@ order: 402
...
 
 ### 针对高级用户
 
-无论你更钟情 webpack 或是 Browserify，我们为简单的和更复杂的项目都提供了一些文档模板。我们建议浏览 [github.com/vuejs-templates](https://github.com/vuejs-templates)，找到你需要的部分，然后参考 README 中的说明，使用 [vue-cli](https://github.com/vuejs/vue-cli) 工具生成新的项目。
+CLI 会为你搞定大多数工具的配置问题，同时也支持细粒度自定义[配置项](https://cli.vuejs.org/config/)。
```
Comment (Justineo) [inline src/v2/guide/single-file-components.md]:
```suggestion
CLI 会为你搞定大多数工具的配置问题，同时也支持细粒度自定义[配置项](https://cli.vuejs.org/zh/config/)。
```

File: src/v2/guide/single-file-components.md [hunk 3]
```diff
@@ -62,12 +62,10 @@ order: 402
...
 
-模板中使用 [webpack](https://webpack.js.org/)，一个模块加载器加载多个模块然后构建成最终应用。为了进一步了解 webpack，可以看 [官方介绍视频](https://www.youtube.com/watch?v=WQue1AN93YU)。如果你有基础，可以看 [在 Egghead.io 上的 webpack 进阶教程](https://egghead.io/courses/using-webpack-for-production-javascript-applications)。
+有时你会想从零搭建你自己的构建工具，这时你需要通过 [vue-loader](https://vue-loader.vuejs.org) 手动配置 webpack。关于学习更多 webpack 的内容，请查阅[其官方文档](https://webpack.js.org/configuration/) 和 [Webpack Academy](https://webpack.academy/p/the-core-concepts)。
```
Comment (Justineo) [inline src/v2/guide/single-file-components.md]:
```suggestion
有时你会想从零搭建你自己的构建工具，这时你需要通过 [Vue Loader](https://vue-loader.vuejs.org/zh/) 手动配置 webpack。关于学习更多 webpack 的内容，请查阅[其官方文档](https://webpack.js.org/configuration/) 和 [Webpack Academy](https://webpack.academy/p/the-core-concepts)。
```

File: src/v2/guide/ssr.md [hunk 4]
```diff
@@ -11,3 +11,7 @@ order: 503
 ## Nuxt.js
 
 从头搭建一个服务端渲染的应用是相当复杂的。幸运的是，我们有一个优秀的社区项目 [Nuxt.js](https://nuxtjs.org/) 让这一切变得非常简单。Nuxt 是一个基于 Vue 生态的更高层的框架，为开发服务端渲染的 Vue 应用提供了极其便利的开发体验。更酷的是，你甚至可以用它来做为静态站生成器。推荐尝试。
+
+## Quasar Framework SSR + PWA
+
+[Quasar Framework](https://quasar-framework.org/) 可以通过其一流的构建系统、合理的配置和开发者扩展性生成 (可选地和 PWA 互通的) SSR 应用，让你的想法的设计和构建变得轻而易举。你可以在服务端挑选执行超过上百款遵循“Material Design 2.0”的组件，并在浏览器端可用。你甚至可以管理网站的 `<meta>` 标签。Quasar 是一个基于 node.js 和 webpack 的开发环境，它可以通过一套代码完成 SPA、PWA、SSR、Electron 和 Cordova 应用的快速开发。
```
Comment (Justineo) [inline src/v2/guide/ssr.md]:
```suggestion
[Quasar Framework](https://quasar-framework.org/) 可以通过其一流的构建系统、合理的配置和开发者扩展性生成 (可选地和 PWA 互通的) SSR 应用，让你的想法的设计和构建变得轻而易举。你可以在服务端挑选执行超过上百款遵循“Material Design 2.0”的组件，并在浏览器端可用。你甚至可以管理网站的 `<meta>` 标签。Quasar 是一个基于 Node.js 和 webpack 的开发环境，它可以通过一套代码完成 SPA、PWA、SSR、Electron 和 Cordova 应用的快速开发。
```

File: src/v2/guide/typescript.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
 ---
 title: TypeScript 支持
 type: guide
-order: 404
+order: 403
 ---
 
-在 Vue 2.5.0 中，我们大大改进了类型声明以更好地使用默认的基于对象的 API。同时此版本也引入了一些其它变化，需要开发者作出相应的升级。阅读[博客文章](https://medium.com/the-vue-point/upcoming-typescript-changes-in-vue-2-5-e9bd7e2ecf08)了解更多详情。
+> [Vue CLI](https://cli.vuejs.org) 内置了 TypeScript 工具支持。在 Vue 的下一个大版本 (3.x) 中也计划了相当多的 TypeScript 支持改进，包括内置的基于 class 的组件 API 和 TSX 的支持。
```
Comment (Justineo) [inline src/v2/guide/typescript.md]:
```suggestion
> [Vue CLI](https://cli.vuejs.org/zh/) 内置了 TypeScript 工具支持。在 Vue 的下一个大版本 (3.x) 中也计划了相当多的 TypeScript 支持改进，包括内置的基于 class 的组件 API 和 TSX 的支持。
```

File: src/v2/guide/unit-testing.md [hunk 6]
```diff
@@ -1,12 +1,10 @@
...
 title: 单元测试
 type: guide
-order: 403
+order: 402
 ---
 
-## 配置和工具
-
-任何兼容基于模块的构建系统都可以正常使用，但如果你需要一个具体的建议，可以使用 [Karma](https://karma-runner.github.io) 进行自动化测试。它有很多社区版的插件，包括对 [Webpack](https://github.com/webpack/karma-webpack) 和 [Browserify](https://github.com/Nikku/karma-browserify) 的支持。更多详细的安装步骤，请参考各项目的安装文档，通过这些 Karma 配置的例子可以快速帮助你上手 ([Webpack](https://github.com/vuejs-templates/webpack/blob/master/template/test/unit/karma.conf.js) 配置，[Browserify](https://github.com/vuejs-templates/browserify/blob/master/template/karma.conf.js) 配置)。
+> [Vue CLI](https://cli.vuejs.org/) 拥有开箱即用的通过 [Jest](https://github.com/facebook/jest) 或 [Mocha](https://mochajs.org/) 进行单元测试的内置选项。我们还有官方的 [Vue Test Utils](https://vue-test-utils.vuejs.org/) 提供更多详细的指引和自定义设置。
```
Comment (Justineo) [inline src/v2/guide/unit-testing.md]:
```suggestion
> [Vue CLI](https://cli.vuejs.org/zh/) 拥有开箱即用的通过 [Jest](https://github.com/facebook/jest) 或 [Mocha](https://mochajs.org/) 进行单元测试的内置选项。我们还有官方的 [Vue Test Utils](https://vue-test-utils.vuejs.org/zh/) 提供更多详细的指引和自定义设置。
```

**Comment by Jinjiang** [comment]
> - https://github.com/vuejs/vuejs.org/commit/0c1ff02161403382e5139e9f4ae56d1e4474e2ba All videos in Vue Mastery are deployed on Vimeo which is blocked in CN. And also the script from vimeo.com would probably block the page loading. So I didn't sync this change. @yyx990803 is that OK to drop this commit here?

> Also, some of the images in `/theme/vue/source/images` has been removed because they seem not useful anymore:
>
> * `coding-net-ad.jpg`
> * `fancygrid.png`
> * `gitee_ad.png`
> * `geekbang.png`
>
> @yyx990803 you may need to have a check because it's sponsors related.

Memo: these two things above have already been orally agreed by @yyx990803. So merged.

**Pull request merged**